crates/mod-crypto/src/lib.rs

```diff
@@ -0,0 +1,375 @@
+                    PaddedBlockDecryptingKey::ecb_pkcs7(key)?
+                }
+                AesKey::Aes256(k) => {
+                    let key = UnboundCipherKey::new(&AES_256, &k)?;
+                    PaddedBlockDecryptingKey::ecb_pkcs7(key)?
+                }
+            };
+
+            match dec_key.decrypt(&mut in_out_buffer, DecryptionContext::None) {
+                //     Ok(DigestResult(digest.as_ref().to_vec()))
+                Ok(plaintext) => Ok(DecryptResult(plaintext.to_vec())),
+                Err(_) => Err(anyhow!("Decryption failed with AES key, mode ecb block")),
+            }
+        }
+
+        AesAlgo::Cbc() => {
+            dec_key = match params.key {
+                AesKey::Aes128(k) => {
+                    let key = UnboundCipherKey::new(&AES_128, &k)?;
+                    PaddedBlockDecryptingKey::cbc_pkcs7(key)?
+                }
+                AesKey::Aes256(k) => {
+                    let key = UnboundCipherKey::new(&AES_256, &k)?;
+                    PaddedBlockDecryptingKey::cbc_pkcs7(key)?
+                }
+            };
+
+            // CBC expects the IV to be prepended to the ciphertext.
+            if ciphertext_buf.len() < 16 {
+                return Err(anyhow!(
+                    "Ciphertext is too short to contain an IV (expected at least 16 bytes for IV)"
+                ));
+            }
+
+            let (iv_bytes, actual_ciphertext) = ciphertext_buf.split_at(16); // Split into IV and actual ciphertext
+            let iv: [u8; 16] = iv_bytes
+                .try_into()
+                .map_err(|_| anyhow!("IV length is incorrect, expected 16 bytes"))?;
+
+            let mut decrypt_buffer = actual_ciphertext.to_vec(); // Create a mutable buffer for decryption
+            let plaintext_slice =
+                dec_key.decrypt(&mut decrypt_buffer, DecryptionContext::Iv128(iv.into()))?;
+            Ok(DecryptResult(plaintext_slice.to_vec()))
+        }
+    }
+}
+pub fn register(lua: &Lua) -> anyhow::Result<()> {
+    let crypto = get_or_create_sub_module(lua, "crypto")?;
+    crypto.set(
+        "aes_encrypt_block",
+        lua.create_async_function(|lua, params: Value| async move {
+            let params: LuaCfg = from_lua_value(&lua, params)?;
+            let aes_k =
+                params.key.get().await.map_err(|e| {
+                    LuaError::external(format!("key: {:?} failed: {}", params.key, e))
+                })?;
+            let aes_key = AesKey::from_bytes(&aes_k).map_err(any_err)?;
+
+            let algo = match params.algorithm.to_lowercase().as_str() {
+                "cbc" => AesAlgo::Cbc(),
+                "ecb" => AesAlgo::Ecb(),
+                _ => {
+                    return Err(LuaError::external(format!(
+                        "Invalid algorithm '{}'. Expected 'cbc' or 'ecb'",
+                        params.algorithm
+                    )));
+                }
+            };
```

Comment:
This whole block should go away once `algorithm` is changed from `String` to `AesAlgo`.

```suggestion
```